src/static/js/app/components/Survey.js

```diff
@@ -0,0 +1,92 @@
+import 'fetch';
+import React from 'react';
+
+import SurveyPage from './SurveyPage';
+
+export default class Survey extends React.Component {
+  state = {
+    done: false,
+    pageHasErrors: false,
+    questions: [],
+    nextPage: 2,
+    userId: 0
+  }
+
+  componentDidMount() {
+    let userId = document.getElementById('user-id').innerHTML;
+    this.getFirstPageForUser(userId);
+  }
+
+  getFirstPageForUser(userId) {
+    fetch(`/api/v1/survey/getFirstPage/?userId=${userId}`)
+      .then((response) => response.json())
+      .then((jsonData) => {
+        this.setState({
+          userId: jsonData.userId
+        });
+        this.updateQuestionsInState(jsonData);
+      });
+  }
+
+  getNextPageOrSubmit = (e) => {
+    if (Object.keys(this._surveyPage.getQuestions()).length !== 20) {
+      this.setState({
+        pageHasErrors: true
+      });
+      return;
+    } else {
+      this.setState({
+        pageHasErrors: false
+      });
+    }
+    this.submitAnswers(this.state.userId);
+    if (this.state.nextPage !== false) {
+      this.getSurveyPage(this.state.userId, this.state.nextPage);
+    } else {
+      // window.location.href = '/results/' + this.state.userId
```

Comment:
I'm still not sure what to do here, especially since I don't wanna answer all 180 questions just to see this! Haha.